### ISSUE TITLE
Does not run from a subdirectory of project in GOPATH 

### ISSUE THREAD

**Issue by OJFord** [go-swagger/go-swagger#887]
## Problem statement

Does not run from a subdirectory in $GOPATH/src.

## Steps to reproduce

```sh
somewhere/on/gopath$ cd subdir && swagger generate server
2017/01/22 06:43:08 building a plan for generation
2017/01/22 06:43:08 target must reside inside a location in the $GOPATH/src
```

## Workaround
```sh
somewhere/on/gopath$ swagger generate server --target=subdir
[generates]
```

I'd use the workaround, but I have the command in `subdir/Makefile` and running with `make -C subdir`.

**Comment by OJFord** [comment]
Oh, actually, not true - what I changed was using the version from `brew install` rather than `go get`.

So it seems it doesn't care where the target is if the binary's in `$GOPATH/bin`?

Why does it need any relationship with `$GOPATH` anyway, at generate-time?

**Comment by casualjim** [comment]
Because we use goimports to resolve imported libraries/packages
You're generating go code, so if you want to compile it, it would also need to be in a gopath
You can compile a single file that just uses stdlib features but for multiple files (like a swagger generated server) that is a very painful situation

**Comment by GlenDC** [comment]
Also it shouldn't make a difference if you use the brew version (installed in `/usr/local/bin`?) or the one from `$GOPATH/bin` though. Can you please give more details in your steps to reproduce this, as the problem might just be that you weren't in a valid $GOPATH, when trying to generate.

**Comment by OJFord** [comment]
Oh, okay.

> You're generating go code, so if you want to compile it, it would also need to be in a gopath

Yep, but I'm compiling it in a Docker container.

> shouldn't make a difference if you use the brew version (installed in /usr/local/bin?)

Yep.

> Can you please give more details in your steps to reproduce this

```sh
> $GOPATH/bin/swagger generate server [...]
[...]
2017/01/22 06:43:08 building a plan for generation
2017/01/22 06:43:08 target must reside inside a location in the $GOPATH/src
> $GOPATH/bin/swagger version
dev
> which swagger
/usr/local/bin/swagger
> swagger version
version: 0.8.0
commit: d1a0b6c1007402b2ccde73702eb0e8e43a5e7498
> swagger generate server [...]
[...]
Generation completed!For this generation to compile you need to have some packages in your GOPATH:

  * github.com/go-openapi/runtime
  * github.com/tylerb/graceful
  * github.com/jessevdk/go-flags
  * golang.org/x/net/context

You can get these now with: go get -u -f ./...
```

**Comment by GlenDC** [comment]
What is your exact working directory (fully specified) at the point of generation?

**Comment by OJFord** [comment]
`~/projectname` - which isn't on my `$GOPATH` now I look at it. But it works despite that with the `go get`ted version.

**Comment by casualjim** [comment]
ic, perhap you can add the swagger binary to your docker container and run the generation in the container itself.

We build for linux as well and we also publish a container with go-swagger already inside based on the official golang container with alpine

**Comment by GlenDC** [comment]
@OJFord you have anymore updates on this one?

**Comment by OJFord** [comment]
I ended up agreeing with @casualjim about running generation in the container, as long as I was able to write my handlers on the host machine without the generated stuff (cf. #888) to avoid needing to synchronise or use a volume.

I haven't actually got that working yet because I ran into #656,  and had to leave it as I don't have time really.

I'll close this though, as it seems I was basically expecting a behaviour to always occur that wasn't intentionally occurring at all.
